### ISSUE TITLE
CRD validation of ingresses

### ISSUE THREAD

**Issue by tronghn** [nais/naiserator#206]
There have been counts of user-errors when defining ingresses for their applications, i.e.:

- setting a non-https scheme: `http://domain.tld` 
- dropping the scheme altogether: `domain.tld`

Both cases result in an error in naiserator when creating/updating resources:

- `creating cluster resource operations: unable to create VirtualServices: URL 'http://domain.tld/' does not start with 'https://'`
- `creating cluster resource operations: while creating ingress: URL 'domain.tld' is missing a hostname`

These errors however aren't readily visible to the end-users during deploys, which is unintuitive and may be confused with deployment pipeline issues.

We should consider adding CRD level validation of the ingresses to return an error at an earlier point, e.g.:

```golang
// +kubebuilder:validation:Pattern=`^https:\/\/`
type Ingress string

// ApplicationSpec contains the NAIS manifest.
type ApplicationSpec struct {
    ...
    Ingresses []Ingress `json:"ingresses,omitempty"`
    ...
}
```

**Comment by jhrv** [comment]
Agree. 

One option might be to allow for defining just the hostname, like we do in the accessPolicies. 

Afaik we only create https ingresses anyway, so to prepend that in the Ingresses/VSes.

This might make this more user friendly.

**Comment by ambientsound** [comment]
> One option might be to allow for defining just the hostname, like we do in the accessPolicies.
IMO it's better to explicitly demand a protocol to make it perfectly clear that this only works on the HTTPS protocol and nothing else. Otherwise users might assume that they can create an ingress to handle other protocols from external callers.

**Comment by jhrv** [comment]
> IMO it's better to explicitly demand a protocol to make it perfectly clear that this only works on the HTTPS protocol and nothing else. Otherwise users might assume that they can create an ingress to handle other protocols from external callers.

👍 

With regards to the accessPolicies, there is already field defining the protocol explicitly there so my original argument is no good.

**Comment by ambientsound** [comment]
Fixed in #230
